plugins/janus_sip.c

```diff
@@ -4374,12 +4374,29 @@ static void *janus_sip_handler(void *data) {
 			record_peer_audio = peer_audio ? json_is_true(peer_audio) : FALSE;
 			json_t *peer_video = json_object_get(root, "peer_video");
 			record_peer_video = peer_video ? json_is_true(peer_video) : FALSE;
-			if(!record_audio && !record_video && !record_peer_audio && !record_peer_video) {
+			gint update_bitmap =
+				record_audio |
+				record_video << 1 |
+				record_peer_audio << 2 |
+				record_peer_video << 3;
+			if(!update_bitmap) {
 				JANUS_LOG(LOG_ERR, "Invalid request (at least one of audio, video, peer_audio and peer_video should be true)\n");
 				error_code = JANUS_SIP_ERROR_RECORDING_ERROR;
 				g_snprintf(error_cause, 512, "Invalid request (at least one of audio, video, peer_audio and peer_video should be true)");
 				goto error;
 			}
+			gint recorder_bitmap =
+				g_atomic_int_get(&session->arc->paused) |
+				g_atomic_int_get(&session->vrc->paused) << 1 |
+				g_atomic_int_get(&session->arc_peer->paused) << 2 |
+				g_atomic_int_get(&session->vrc_peer->paused) << 3;
+			if ((!strcasecmp(action_text, "pause") && !(~recorder_bitmap & update_bitmap)) ||
+				(!strcasecmp(action_text, "resume") && !(recorder_bitmap & update_bitmap))) {
+				JANUS_LOG(LOG_ERR, "Invalid pause/resume request (at least one of audio, video, peer_audio and peer_video should change recorder state)\n");
```

Comment:
I don't think this bitmap approach works, which feels a bit convoluted. First all, this will crash Janus if you're not recording everything (e.g., just the user, just the peer, or video is not active), as in that case some recordings object would be `NULL`. Besides, I suspect this check would break if the provided requests are not aligned with the state, and go in an unrecoverable state.

Considering the state enhancement you added to `record.c` that does make the pause/resume methods idempotent, I think we can keep the code as it was in the SIP plugin for the sake of simplicity, but only send the PLI if we're really resuming (so only adding a single additional check below).